### PR TITLE
identify server by FQDN

### LINKED ISSUE #79: [dist] allow FQDN for server/scheduler nodes in the config files
Currently only IP addresses are accepted, but we really should accept FQDN for use with kubernetes or nomad or service meshes in general.

### PR DESCRIPTION
**Pull request by montekki**
Fixes #79

### CODE REVIEW

File: src/bin/cachepot-dist/main.rs [hunk 1]
```diff
@@ -90,7 +89,7 @@ struct GenerateJwtHS256ServerToken {
 
     /// Generate a key for the specified server
     #[structopt(long, value_name = "SERVER_ADDR", required_unless = "secret_key")]
-    server: SocketAddr,
+    server: String,
```
Comment (drahnr) [inline src/bin/cachepot-dist/main.rs:92]: `#[structopt(parse(from_os_str))]` should avoid the deferring parsing I'd assume?

**Comment by drahnr** [comment]
Given this, we can still use the IP variant, correct? So instead of `http://192.168.111.111:7777` would still work as expected?

File: src/bin/cachepot-dist/main.rs (hunk 1 above)
Comment (montekki) [reply src/bin/cachepot-dist/main.rs]: right

**Comment by montekki** [comment]
For all server-id related things you can use ip as before (`127.0.0.1:10600`) or name (`localhost:10600`) and this is compatible with prev versions of configs and API. Like the comment to `ServerUrl` states one of the reasons for dropping the `https://` scheme is that we would have to additionally urlencode params in the api calls.

**Review by drahnr — APPROVED** [review]
The change looks good to me

**Pull request merged**